src/commands/utils/roleAssignments.ts

```diff
@@ -0,0 +1,128 @@
+}
+
+// There are several permitted principal types, see: https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/create?view=rest-authorization-2022-04-01&tabs=HTTP#principaltype
+// For now, 'ServicePrincipal' and 'User' are the ones we're most likely to use here,
+// but we can add more as needed.
+export type PrincipalType = "ServicePrincipal" | "User";
+
+export async function createRoleAssignment(
+    client: AuthorizationManagementClient,
+    subscriptionId: string,
+    principalId: string,
+    roleDefinitionName: string,
+    principalType: PrincipalType,
+    scope: string,
+): Promise<Errorable<RoleAssignment>> {
+    const newRoleAssignmentName = createRoleAssignmentName();
+    const roleDefinitionId = `/subscriptions/${subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${roleDefinitionName}`;
+
+    const newRoleAssignment: RoleAssignmentCreateParameters = {
+        principalId,
+        roleDefinitionId,
+        principalType,
+    };
+
+    try {
+        const roleAssignment: RoleAssignment = await client.roleAssignments.create(
+            scope,
+            newRoleAssignmentName,
+            newRoleAssignment,
+        );
+        return { succeeded: true, result: roleAssignment };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function deleteRoleAssignment(
+    client: AuthorizationManagementClient,
+    subscriptionId: string,
+    principalId: string,
+    roleDefinitionName: string,
+    scope: string,
+): Promise<Errorable<void>> {
+    const roleDefinitionId = `/subscriptions/${subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${roleDefinitionName}`;
+
+    const roleAssignments = await listAll(client.roleAssignments.listForScope(scope));
+    if (failed(roleAssignments)) {
+        // Returning the inner error propagates it up the call chain.
+        return roleAssignments;
+    }
+
+    const roleAssignment = roleAssignments.result.find(
+        (ra) => ra.principalId === principalId && ra.roleDefinitionId === roleDefinitionId,
+    );
+
+    if (!roleAssignment) {
+        return { succeeded: true, result: undefined };
+    }
+
+    if (!roleAssignment.id) {
+        return { succeeded: false, error: "Role assignment has no ID" };
+    }
+
+    try {
+        await client.roleAssignments.deleteById(roleAssignment.id);
+        return { succeeded: true, result: undefined };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+function createRoleAssignmentName(): string {
+    // https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments#name
+    // "A role assignment's resource name must be a globally unique identifier"
+    // https://stackoverflow.com/a/2117523
+    return "10000000-1000-4000-8000-100000000000".replace(/[018]/g, (c) =>
```

Comment:
In general I'm not keen on adding another dependency (in this case two - because it needs TypeScript types) for a single short function...but no strong feelings and the dependency you suggest seems well-trusted and innocuous, so I've made that change.